iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/unique.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include <utility>
+
+namespace iox
+{
+template <typename T>
+class unique
+{
+  public:
+    struct invalid_t
+    {
+        explicit constexpr invalid_t()
+        {
+        }
+    };
+
+    static constexpr invalid_t invalid{};
+
+    template <typename... Args>
+    unique(Args&&... args)
+        : m_value(std::forward<Args>(args)...)
+    {
+    }
+
+    unique()
+        : m_value()
+    {
+    }
+
+    unique(invalid_t)
+        : m_valid(false)
+    {
+    }
+
+    // no copies
+    unique(const unique&) = delete;
+    unique& operator=(const unique&) = delete;
+
+    // only move
+    unique(unique&& other)
+        : m_value(std::move(other.m_value))
+        , m_valid(other.m_valid)
+    {
+        other.m_valid = false;
+    };
+
+    unique& operator=(unique&& other)
+    {
+        if (this != &other)
+        {
+            m_value = std::move(other.m_value);
+            m_valid = other.m_valid;
+            other.m_valid = false;
+        }
+        return *this;
+    }
+
+    const T& operator*() const
+    {
+        return m_value;
+    }
+
+    T release()
+    {
+        m_valid = false;
+        return std::move(m_value);
+    }
+
+    operator const T&() const
+    {
+        return m_value;
+    }
+
+    bool isValid() const
+    {
+        return m_valid;
+    }
+
+    // note: interferes with operator const T& for some types, we cannot have both(which do we want ?)
+    // operator bool()
```

Comment:
I think this operator would be always wrong. Let's say you have a ```unique<bool> a(false)``` but then you use it like that
```
if ( a ) {
  printf("true");
} else {
  printf("false");
}
```
It would print false even though the ```unique<bool> a``` is true since the bool returns if its set or not and this is very confusing.
I would leave it as it is and remove this note

Reply:
Yeah, this is just a leftover. I mainly wanted to have the object behave like the underlying type T, using implicit conversion. It is also intended to be used with small types T, otherwise this gets expensive. To be true, this abstraction does more harm than good (performance and maintainance-wise)